jujugui/static/gui/src/app/app.js

```diff
@@ -1933,6 +1933,16 @@ YUI.add('juju-gui', function(Y) {
       if (this.get('sandbox')) {
         console.log('switching models is not supported in sandbox');
       }
+      // Optionally update to the new base URL.
+      if (this.get('gisf')) {
+        var state = this.state;
+        var baseUrl = state.get('baseUrl');
+        var unsavedModelUrl = 'sandbox';
+        // Make sure we're not already in an unsaved model.
+        if (baseUrl.indexOf(unsavedModelUrl) < 0) {
```

Comment:
This seems like an unsafe way of telling if the model is unsaved. Do we not have a specific flag for this? What if a user called there real model sandbox?